compiler/rustc_data_structures/src/sharded.rs

```diff
@@ -142,9 +142,13 @@ fn make_hash<K: Hash + ?Sized>(val: &K) -> u64 {
 /// consistently for each `Sharded` instance.
 #[inline]
 pub fn get_shard_index_by_hash(hash: u64) -> usize {
-    let hash_len = mem::size_of::<usize>();
-    // Ignore the top 7 bits as hashbrown uses these and get the next SHARD_BITS highest bits.
-    // hashbrown also uses the lowest bits, so we can't use those
-    let bits = (hash >> (hash_len * 8 - 7 - SHARD_BITS)) as usize;
-    bits % SHARDS
+    if SHARDS == 1 {
```

Comment:
https://godbolt.org/z/cq9nMvz1q This change is maybe redudant, since the optimizer will see that `bits % 1` has to equal zero?